add 3dmol.js post

duerrsimon opened the pull request:
I wrote a short post on how to visualize proteins using gradio as discussed with @osanseviero 

Let me know what you think :)

osanseviero commented:
As discussed in a DM, we'll review it here first and once ready we'll have a Gradio Guide with same content. I will review this Friday of next week but maybe @abidlabs is interested in reviewing as well! :partying_face: 

Thank you for the contribution!!

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+thumbnail: /blog/assets/94_visualize_proteins_using_3dmoljs/thumbnail.png
+---
+
+<h1>Visualize proteins on Huggingface spaces</h1>
```
```suggestion
<h1>Visualize proteins on Hugging Face Spaces</h1>

```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+---
```
You'll also need to add the blog entry to https://github.com/huggingface/blog/blob/main/_blog.yml for it to be shown

duerrsimon replied:
I used only the research tag. Not sure if you want add one related to protein ML

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+<h1>Visualize proteins on Huggingface spaces</h1>
+
+<div class="blog-metadata">
+    <small>Published 10.08, 2022.</small>
```
```suggestion
    <small>Published August 24, 2022.</small>
```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+
+<div class="blog-metadata">
+    <small>Published 10.08, 2022.</small>
+    <a target="_blank" class="btn no-underline text-sm mb-5 font-sans" href="https://github.com/huggingface/blog/blob/main/pytorch-fsdp.md">
```
Please update the path to the markdown

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+    </a>
+</div>
+
+In this post we will look at how we can visualize proteins on Huggingface spaces.
```
```suggestion
In this post we will look at how we can visualize proteins on Hugging Face Spaces.

```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+        <img class="avatar avatar-user" src="hhttps://avatars.githubusercontent.com/u/16979577?s=200&v=4" title="Gravatar">
+        <div class="bfc">
+            <code>duerrsimon</code>
+            <span class="fullname">Simon Dürr</span>
```
Could you add a guest tag as in https://github.com/merveenoyan/blog/blob/9fac04883b50f948e75065e51d1396bdc23bbfa4/deploy-tfserving-kubernetes.md?plain=1#L31?

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+
+## Seeing is believing
+
+The structure of a protein is an integral part to our understanding of what a protein does. Nowadays, there are a few tools available to visualize proteins directly in the browser such as [mol*](molstar.org) or [3dmol.js](https://3dmol.csb.pitt.edu/). In this post, you will learn how to integrate structure visualization into your Huggingface space using 3Dmol.js and the HTML block. 
```
```suggestion
The structure of a protein is an integral part to our understanding of what a protein does. Nowadays, there are a few tools available to visualize proteins directly in the browser such as [mol*](molstar.org) or [3dmol.js](https://3dmol.csb.pitt.edu/). In this post, you will learn how to integrate structure visualization into your Hugging Face Space using 3Dmol.js and the HTML block. 

```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+
+Let's take a look at how to create the minimal working demo of our interface before we dive into how to setup 3Dmol.js. 
+
+We will built a simple demo app that can accept either a 4-digit PDB code  or a PDB file. Our app will then retrieve the pdb file from the RCSB Protein Databank and display it or use the uploaded file for display.
```
```suggestion
We will build a simple demo app that can accept either a 4-digit PDB code or a PDB file. Our app will then retrieve the pdb file from the RCSB Protein Databank and display it or use the uploaded file for display.

```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+
+We will built a simple demo app that can accept either a 4-digit PDB code  or a PDB file. Our app will then retrieve the pdb file from the RCSB Protein Databank and display it or use the uploaded file for display.
+
+<iframe src="https://hf.space/embed/simonduerr/3dmol.js/+
```
Could we embed the space with https://gradio.app/sharing_your_app/#embedding-with-web-components? It has the benefit of being responsive and nicer code :) You can add the script part here

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+import gradio as gr
+
+def update(inp, file):
+    # in this simple example we just retrieve the pdb file using its identifier from the RCSB or display the uploaded  file
```
```suggestion
    # in this simple example we just retrieve the pdb file using its identifier from the RCSB or display the uploaded file

```

osanseviero commented on visualize_proteins_3dmol.js.md:
```diff
@@ -0,0 +1,168 @@
+    mol = read_mol(input_pdb)
+    # setup HTML document
+    x = ("""<!DOCTYPE html><html> [..] </html>""") # do not use ' in this input
+    return f"""<iframe  [..] srcdoc='{x}'></iframe>
```
Starting this line I see lots of `^M` characters. Is this something we could remove?

duerrsimon replied:
I fixed it now using dos2unix

osanseviero commented on visualize_proteins_3dmol.js.md:
````diff
@@ -0,0 +1,168 @@
+```
+We use a template literal (denoted by backticks) to store our pdb file in the html document directly and then output it using 3dmol.js.
+
+And that's it, now you can couple your favorite protein ML model to a fun and easy to use gradio app and directly visualize predicted or redesigned structures. If you are predicting properities of a structure (e.g how likely each amino acid is to bind a ligand) 3Dmol.js also allows to use custom `colorfunc` based on a property of each atom. 
````
```suggestion
And that's it, now you can couple your favorite protein ML model to a fun and easy to use gradio app and directly visualize predicted or redesigned structures. If you are predicting properities of a structure (e.g how likely each amino acid is to bind a ligand), 3Dmol.js also allows to use a custom `colorfunc` based on a property of each atom. 

```

duerrsimon replied:
I copy pasted the code on my windows machine and that must have inserted these line endings. I don't see them in VS Code and they only show up in the github soure. I'll get this fixed

osanseviero reviewed (approved):
Very cool blog post! Thanks a lot for writing it! I did a first pass but this is already looking in a great direction :fire:

duerrsimon commented:
All requested changes have been incorporated. I also removed the duplicate id for the post (there was already a folder assets/94_xx so I changed it to the first free number (98).

osanseviero commented:
Awesome, amazing work :) I'll merge+deploy this tomorrow
